### ISSUE TITLE
Rubocop warns that department needs migration, but doesn't migrate

### ISSUE THREAD

**Issue by tas50** [rubocop-hq/rubocop#7627]
## Expected behavior

When RuboCop warns the user that no department name is given and that the Migration cop needs to be run it should successfully migrate the cop comments.

## Actual behavior

I have a bit of ruby code that contains a rubocop disable comment like this:

```ruby
    # rubocop : disable CyclomaticComplexity
    def merge(predicate, hash1, force = false)
      raise "the predicate should be of type: boolean, not: #{predicate.class.name}" unless [true, false].include?(predicate)
      return if hash1.nil?
      @hash = @hash.merge(hash1) if predicate && !force
      @hash = @hash.merge!(hash1) if predicate && force
    end
    # rubocop : enable CyclomaticComplexity
```

When running rubocop these legacy department-less comments produce a warning:

```
Inspecting 1 file
/Users/tsmith/dev/the_world/mapr/libraries/mapr_utils.rb: Warning: no department given for CyclomaticComplexity. Run `rubocop -a --only Migration/DepartmentName` to fix.
/Users/tsmith/dev/the_world/mapr/libraries/mapr_utils.rb: Warning: no department given for CyclomaticComplexity. Run `rubocop -a --only Migration/DepartmentName` to fix.
.
```

The problem is that running this command produces the same output:

```
rubocop -a --only Migration/DepartmentName /Users/tsmith/dev/the_world/mapr/libraries/mapr_utils.rb
Inspecting 1 file
/Users/tsmith/dev/the_world/mapr/libraries/mapr_utils.rb: Warning: no department given for CyclomaticComplexity. Run `rubocop -a --only Migration/DepartmentName` to fix.
/Users/tsmith/dev/the_world/mapr/libraries/mapr_utils.rb: Warning: no department given for CyclomaticComplexity. Run `rubocop -a --only Migration/DepartmentName` to fix.
```

## RuboCop version

```
0.78.0
```

**Comment by pboling** [comment]
This is still (or again?) happening on latest release, 1.44.1

```
$  bundle exec rubocop -v
1.44.1

$ bundle exec rubocop -a --only Migration/DepartmentName
/Users/pboling/src/.../client.rb: Warning: no department given for RescueException. Run `rubocop -a --only Migration/DepartmentName` to fix.
/Users/pboling/src/.../client.rb: Warning: no department given for RescueException. Run `rubocop -a --only Migration/DepartmentName` to fix.
/Users/pboling/src/.../client.rb: Warning: no department given for RescueException. Run `rubocop -a --only Migration/DepartmentName` to fix.
/Users/pboling/src/.../client.rb: Warning: no department given for RescueException. Run `rubocop -a --only Migration/DepartmentName` to fix.
/Users/pboling/src/.../listener.rb: Warning: no department given for RescueException. Run `rubocop -a --only Migration/DepartmentName` to fix.
/Users/pboling/src/.../listener.rb: Warning: no department given for RescueException. Run `rubocop -a --only Migration/DepartmentName` to fix.
/Users/pboling/src/.../listener.rb: Warning: no department given for RescueException. Run `rubocop -a --only Migration/DepartmentName` to fix.
/Users/pboling/src/.../listener.rb: Warning: no department given for RescueException. Run `rubocop -a --only Migration/DepartmentName` to fix.
Inspecting 515 files
...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

515 files inspected, no offenses detected
```

It doesn't fix the offense, despite saying it will.
